cleanlab/experimental/datalab/data.py

```diff
@@ -0,0 +1,217 @@
+        """Checks the type of dataset and uses the correct loader method and
+        assigns the result to the data attribute."""
+        if isinstance(data, datasets.Dataset):
+            return data
+        if isinstance(data, pd.DataFrame):
+            return datasets.Dataset.from_pandas(data)
+
+        if isinstance(data, dict):
+            try:
+                return datasets.Dataset.from_dict(data)
+            except Exception as e:
+                raise ValueError(
+                    "Failed to load data from dictionary. "
+                    "Ensure that it is a mapping from strings "
+                    "to arrays or lists of equal length.\n"
+                    f"Error: {e}"
+                ) from e
+        if isinstance(data, list):
+            try:
+                return datasets.Dataset.from_list(data)
+            except Exception as e:
+                raise ValueError(
+                    "Failed to load data from list. "
+                    "Ensure that it is a list of dictionaries "
+                    "with the same keys.\n"
+                    f"Error: {e}"
+                ) from e
+
+        if isinstance(data, str):
+            # Check if it's a path to a local file
+            import os
+
+            if os.path.exists(data):
+                if data.endswith(".txt"):
+                    return datasets.Dataset.from_text(data)
+                if data.endswith(".csv"):
+                    return datasets.Dataset.from_csv(data)
+                if data.endswith(".json"):
+                    return datasets.Dataset.from_json(data)
+
+            return datasets.load_dataset(data)
+
+    def __len__(self) -> int:
+        return len(self._data)
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, Data):
+            # Equality checks
+            hashes = self._data_hash == other._data_hash
+            labels = np.array_equal(self._labels, other._labels)
+            label_names = self._label_name == other._label_name
+            label_maps = self._label_map == other._label_map
+            return all([hashes, labels, label_names, label_maps])
+        return False
+
+    def __hash__(self) -> int:
+        return self._data_hash
+
+    @property
+    def class_names(self) -> list:
+        label_name = cast(str, self._label_name)
+        return self._data.unique(label_name)
+
+    @staticmethod
+    def _validate_data(data) -> None:
+        if isinstance(data, datasets.DatasetDict):
+            raise ValueError(
+                "Please pass a single dataset, not a DatasetDict. "
+                "Try initializing with data['train'] instead."
+            )
+
+        assert isinstance(data, Dataset)
+
+    @staticmethod
+    def _validate_data_and_labels(data, labels) -> None:
+        if isinstance(labels, np.ndarray):
+            assert labels.shape[0] == data.shape[0]
+
+        if isinstance(labels, str):
+            pass
```

Comment:
Related to @sanjanag's comment above, I think something is incorrect here - I'm also not sure why there would be a case where `labels` would be of type `str`?
I see a comment above that it will be updated but leaving this comment here as a reminder.